b9lab-content/3-main-concepts/05-transactions.md

```diff
@@ -1 +1,162 @@
-# Transactions
\ No newline at end of file
+---
+title: "Transactions"
+order: 5
+description: Transactions in Cosmos.
+---
+
+# Transactions
+
+Transactions are objects created by end-users to trigger state changes in the application. They are comprised of metadata that defines a context and `sdkMsgs` that trigger state changes within a module through the module’s Protobuf message service.
+
+## Transaction Process, End-user
+
+While there is much to explore as we journey through the stack, let us first describe the transaction process from a user perspective.
+
+* **Decide** on the messages to put into the transaction. This is normally done with the assistance of a wallet or application and a user interface.
+* **Generate** the transaction using the SDK's `TxBuilder`. `TxBuilder` is the preferred way to generate a transaction.
+* **Sign** the transaction. Transactions must be signed before the validators will include them in a block.
+* **Broadcast** the signed transaction using one of the available interfaces.
+
+From a user perspective, **decide** and **sign** are the main interactions, while **generate** and **broadcast** are attended by the user interface and other automations.
+
+## Transaction Objects
+
+Transaction objects are SDK types that implement the `Tx` interface. They contain the following methods:
```

Comment:
I think this doesn't reflect how SDK devs think about txs, and this confusion comes from legacy code. We created the `Tx` interface because we had to support 2 types of tx: the proto Tx, and the amino StdTx. In practice, not one single chain (afaik) uses amino StdTx, so the transaction object is really the proto tx: https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/tx/v1beta1/tx.proto#L13-L26 (also see a [JSON representation](https://api.cosmos.network/cosmos/tx/v1beta1/txs/D5F3927B9BCE4F429155B60E626FCDDBC39FC078C435B038BAE358D51FB1A494)).

My suggestion here would be to show the schema of the proto transaction object, and mention that it implements the `Tx` interface with the 2 methods below.